Handle importing errors

Puigcerber opened the pull request:
Do import returns an array of objects with the status property set to false if it failed to import, and 
 true otherwise. 

I believe it will never reach the `catch` and therefore import errors need to be handle in a different way.

<img width="1030" alt="screen shot 2017-08-24 at 10 57 15" src="https://user-images.githubusercontent.com/866808/29658755-56cb3052-88bc-11e7-83f6-26f5286cb0a5.png">

Voles commented on src/lib/Watcher.js:
```diff
@@ -113,17 +113,24 @@ const runPackageAndImport = widgetDirectory => {
 		.then(() => _import.doImport())
-		.then(() => notifySuccess())
-		.catch(error => notifyError(error));
+		.then((sequence) => {
```
Can be written as `.then(sequence => {`

Voles commented on src/lib/Watcher.js:
```diff
@@ -113,17 +113,24 @@ const runPackageAndImport = widgetDirectory => {
-		.then(() => notifySuccess())
-		.catch(error => notifyError(error));
+		.then((sequence) => {
+			let errors = sequence.filter(element => element.status === false).map(element => element.response);
```
I suggest to improve the readability here by using the following notation:
```
const errors = sequence
    .filter(element => element.status === false)
    .map(element => element.response);
```

Voles replied:
Note I've used `const` instead of `let`.

Voles commented on src/lib/Watcher.js:
```diff
@@ -113,17 +113,24 @@ const runPackageAndImport = widgetDirectory => {
  */
-const notifyError = (error) => {
-	Notify.log(error);
+const notifyError = (errors) => {
```
As the function now accepts multiple errors, it makes more sense to rename it to `notifyErrors`.

Voles reviewed (dismissed):
@Puigcerber first of all, thanks for your contribution! Catching the error's is something we've tried to figure out as well but have not yet figured out.

"Do import returns an array of objects with the status property set to false if it failed to import, and
true otherwise." I'm not sure if this is a question or a remark?

1. Can you remove the generated files in the `/dist` folder? We'll generate these when a new release is being made.
2. The error being shown is not telling enough in my opinion. Do you have any suggestions on how we can improve on that?

Voles commented:
This PR solves #18

Puigcerber commented:
Hey @Voles! 

I'm not sure if it solves #18.

"Do import returns an array of objects with the status property set to false if it failed to import, and
true otherwise." was a remark referred to the `do-import.js` file of the bb-import.

The errors are printed anyway by the output of the bb-import, so we could make bb-watch just to print in red that the import failed.

nickthesing commented:
It doesn't solve #18, removed that comment. I'm fine with your suggestion to make it red when it fails. But we can add a custom error message when you encounter that error from bb-import. This way we provide the user a clean solution and explain its not the watcher that is failing but bb-import.

Puigcerber committed:
Handle importing errors

Do import returns an array of objects with the status property set to
false if it failed to import, and true otherwise.

Puigcerber commented:
@Voles @nickthesing I've updated the PR with the required changes. Sorry for the delay.

Voles commented:
@Puigcerber thanks very much for your contribution! 🌻
